examples/pytorch/instance-segmentation/run_instance_segmentation.py

```diff
@@ -0,0 +1,489 @@
+    return batch
+
+
+@dataclass
+class ModelOutput:
+    class_queries_logits: torch.Tensor
+    masks_queries_logits: torch.Tensor
+
+
+def nested_cpu(tensors):
+    if isinstance(tensors, (list, tuple)):
+        return type(tensors)(nested_cpu(t) for t in tensors)
+    elif isinstance(tensors, Mapping):
+        return type(tensors)({k: nested_cpu(t) for k, t in tensors.items()})
+    elif isinstance(tensors, torch.Tensor):
+        return tensors.cpu().detach()
+    else:
+        return tensors
+
+
+class Evaluator:
+    """
+    Compute metrics for the instance segmentation task.
+    """
+
+    def __init__(
+        self,
+        image_processor: Union[MaskFormerImageProcessor, Mask2FormerImageProcessor],
+        id2label: Mapping[int, str],
+        threshold: float = 0.0,
+    ):
+        """
+        Initialize evaluator with image processor, id2label mapping and threshold for filtering predictions.
+
+        Args:
+            image_processor (Union[MaskFormerImageProcessor, Mask2FormerImageProcessor]): Image processor for
+                `post_process_instance_segmentation` method.
+            id2label (Mapping[int, str]): Mapping from class id to class name.
+            threshold (float): Threshold to filter predicted boxes by confidence. Defaults to 0.0.
+        """
+        self.image_processor = image_processor
+        self.id2label = id2label
+        self.threshold = threshold
+        self.metric = self.get_metric()
+
+    def get_metric(self):
+        metric = MeanAveragePrecision(iou_type="segm", class_metrics=True)
+        return metric
+
+    def reset_metric(self):
+        self.metric.reset()
+
+    def postprocess_target_batch(self, target_batch) -> List[Dict[str, torch.Tensor]]:
+        """Collect targets in a form of list of dictionaries with keys "masks", "labels"."""
+        batch_masks = target_batch[0]
+        batch_labels = target_batch[1]
+        post_processed_targets = []
+        for masks, labels in zip(batch_masks, batch_labels):
+            post_processed_targets.append(
+                {
+                    "masks": masks.to(dtype=torch.bool),
+                    "labels": labels,
+                }
+            )
+        return post_processed_targets
+
+    def get_target_sizes(self, post_processed_targets) -> List[List[int]]:
+        target_sizes = []
+        for target in post_processed_targets:
+            target_sizes.append(target["masks"].shape[-2:])
+        return target_sizes
+
+    def postprocess_prediction_batch(self, prediction_batch, target_sizes) -> List[Dict[str, torch.Tensor]]:
+        """Collect predictions in a form of list of dictionaries with keys "masks", "labels", "scores"."""
+
+        model_output = ModelOutput(class_queries_logits=prediction_batch[0], masks_queries_logits=prediction_batch[1])
+        post_processed_output = self.image_processor.post_process_instance_segmentation(
```

Comment:
Hi @SangbumChoi , thanks a lot for taking time to look into this  🤗 
Not sure I got your question, could you provide more details?

Reply:
when the image size is very large such as 2000x3000, I faced this kind of process (such as resizing into original target size) took almost 10s for single image.

I think I need to test this script with large file :)

Reply:
Ok, got it. I observed that evaluation is relatively slow. However, we have to apply postprocessing in order to get metrics. In case it is too slow we can probably set a rare evaluation strategy (specify a large number of steps between evaluations) or completely drop evaluation during the training and rely on loss values only.
Another thing is to test different approaches to speed up this step, e.g. use lower resolution predicted masks, compute on GPU, or find more optimized operations for postprocessing.
Any help and feedback is appreciated 🤗

Reply:
Yeah, that is also what I thought. I just wanted to make sure if there is any methods in find more optimized operations for postprocessing. Thanks for sharing :)